src/WebJobs.Extensions.DurableTask/Correlation/TraceContextBaseExtensions.cs

```diff
@@ -16,16 +16,26 @@ public static class TraceContextBaseExtensions
         /// Create RequestTelemetry from the TraceContext.
         /// </summary>
         /// <param name="context">TraceContext.</param>
+        /// <param name="nameResolver">NameResolver.</param>
         /// <returns>RequestTelemetry.</returns>
-        public static RequestTelemetry CreateRequestTelemetry(this TraceContextBase context)
+        public static RequestTelemetry CreateRequestTelemetry(this TraceContextBase context, INameResolver nameResolver)
         {
             var telemetry = new RequestTelemetry { Name = context.OperationName };
             telemetry.Duration = context.Duration;
             telemetry.Timestamp = context.StartTime;
             telemetry.Id = context.TelemetryId;
             telemetry.Context.Operation.Id = context.TelemetryContextOperationId;
+
+            // Operation Name is set as "DtActivity FunctionName" and we only want the "FunctionName" as the operationName
+            string[] operationNameArray = context.OperationName.Split(' ');
+            bool isActivityRequest = operationNameArray.Length == 2 && string.Equals(operationNameArray[0], TraceConstants.Activity);
+            telemetry.Context.Operation.Name = isActivityRequest ? operationNameArray[1] : context.OperationName;
+
             telemetry.Context.Operation.ParentId = context.TelemetryContextOperationParentId;
 
+            var resolvedSiteName = nameResolver?.Resolve("WEBSITE_SITE_NAME")?.ToLower();
+            telemetry.Context.Cloud.RoleName = resolvedSiteName != null ? resolvedSiteName : string.Empty;
```

Comment:
nit: Can simplify the right side of this assignment with `resolvedSiteName ?? string.Empty`